src/pybind/cephfs/cephfs.pyx

```diff
@@ -1097,7 +1104,7 @@ cdef class LibCephFS(object):
             raise make_ex(ret, "error in setxattr")
 
 
-    def stat(self, path, follow_symlink=True):
+    def stat(self, path, follow_symlink=True, basic=True):
```

Comment:
The idea behind statx is that it is extensible in the future. So I'm not sure a binary switch like this is a good thing to do. What if we add some new per-inode attribute and extend ceph_statx in the future with a new CEPH_STATX_* flag? Should I expect to get the new value in that case, or will we need an "extended" flag in the future?

Also, I'll note that this interface is suboptimal in another way too. Extending stat in this way doesn't allow you to be selective about what fields you want the program to fetch.

One of the design principles behind statx is that it allows the caller to indicate what fields are of interest. For example, if the caller just wants to know the mode of the file, stat() doesn't allow you to query that without also fetching the size and mtime. On a filesystem like cephfs, that has real performance impacts. Fetching the size and mtime may require the MDS to revoke file caps from a client that is actively writing. As a real-world example, this is the cover letter for a patchset to the "ls" command that has some numbers.

https://lists.gnu.org/archive/html/coreutils/2019-09/msg00032.html

Again, I think the best long-term thing you could do would be to just leave the stat method alone here, and add a new statx method that allows for all of this functionality.

Reply:
@jtlayton 
Sorry, I',m later
when it comes to stats to get statistics, the performance problem caused by the caps consistency requirement gives a good hint. thank you!
I have changed to a single mask to get attr and ext information, please reviews again.

Reply:
If there is no problem with the implementation, the qa code will be submitted later.